pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -250,27 +250,40 @@ func (s *Service) ReconcileLoadBalancerMember(openStackCluster *infrav1.OpenStac
 	return nil
 }
 
-func (s *Service) DeleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, loadBalancerName string) error {
+func (s *Service) DeleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
+	loadBalancerName := getLoadBalancerName(clusterName)
 	lb, err := checkIfLbExists(s.loadbalancerClient, loadBalancerName)
 	if err != nil {
 		return err
 	}
-	if lb == nil {
-		// nothing to do
-		return nil
-	}
 
-	deleteOpts := loadbalancers.DeleteOpts{
-		Cascade: true,
-	}
-	s.logger.Info("Deleting load balancer", "name", loadBalancerName, "cascade", deleteOpts.Cascade)
-	err = loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
-	if err != nil {
-		record.Warnf(openStackCluster, "FailedDeleteLoadBalancer", "Failed to delete load balancer %s with id %s: %v", lb.Name, lb.ID, err)
-		return err
-	}
+	if lb != nil && lb.VipPortID != "" {
```

Comment:
is this really correct?
I get that we only want to disassociate the floating ip and delete it if lb.VipPortID != "", but I think we want to delete the lb even if the floating ip is already gone

Reply:
Updated, review again, please.